test

```diff
@@ -96,22 +96,40 @@ function cov_pass {
 
 	mkdir -p "$COVERDIR"
 
+	# convert COVERDIR to absolute path
+	# e2e needs absolute path to put its coverfiles
+	export COVERDIR=$(get_abs_filename "$COVERDIR")
+
 	# PKGS_DELIM contains all the core etcd pkgs delimited by ',' which will be profiled for code coverage.
 	# Integration tests will generate code coverage for those pkgs 
 	PKGS_DELIM=$(echo $TEST | sed 's/ /,/g')
 
-	# TODO create coverage to e2e test
 	PKGS=`echo "$TEST_PKGS" | egrep -v "(e2e|functional-tester)"`
-
+	# run code coverage for unit and integration tests
 	for t in ${PKGS}; do
 		tf=`echo $t | tr / _`
 		#  uses -run=Test to skip examples because clientv3/ example tests will leak goroutines
 		go test -covermode=set -coverpkg $PKGS_DELIM -timeout 15m -run=Test -v -coverprofile "$COVERDIR/${tf}.coverprofile"  ${REPO_PATH}/$t
 	done
 
+	# run code coverage for e2e tests
+	if [ ! -f "bin/etcd_test" ]; then
```

Comment:
this check should be before any tests run for the pass; otherwise it can get half way through then fail because there wasn't a binary in the first place